### ISSUE TITLE
s3fs fails to build where osxfuse is installed (dependent on fuse4osx rather than osxfuse)

### ISSUE THREAD

**Issue by johntrandall** [Homebrew/homebrew#36722]
fuse4x has been long abandoned and merged into osxfuse. (see http://fuse4x.github.io/).

The s3fs formula lists fuse4x as a dependancy. This isn't viable, because fuse4x conflicts with osxfuse. Can the s3fs formula be tweaked to use osxfuse instead? 

Apologies in advance if this has an obvious answer or if I'm missing something...

**Comment by adamv** [comment]
Does it work if this change is made? Can you make a pull request for it?

**Comment by johntrandall** [comment]
I'm not sure I have the knowledge to test if it works accurately. I'm just getting into playing around with fuse. I'm very much a newb.

**Comment by DomT4** [comment]
It doesn't build because upstream have hardcoded in a ` fuse >= 2.8.4 ` requirement. It needs to be kicked upstream really. From a brief glance it looks like the source code is targeted towards Linux distros - Debian Wheezy, the oldest FUSE supporting Debian for example, currently ships FUSE 2.9.0.

Osxfuse currently define their version of FUSE as ` 2.7.3 `, hence the failure.

**Comment by johntrandall** [comment]
I see. Well, could this be pinned in homebrew to the latest version that works with OSX/osxfuse then?

**Comment by DomT4** [comment]
To my knowledge, ` s3fs ` has never successfully compiled against Osxfuse on OS X. I did some light patching of the source code checks locally, but it seems there are actual meaningful changes behind the FUSE version differences and the ` s3fs ` configure/make checks on the FUSE version - Whether that's an OS X Vs Linux change or a change made upstream by FUSE I don't know enough to say.

```
s3fs.cpp:2677:26: error: no member named 'capable' in 'fuse_conn_info'
  if((unsigned int)conn->capable & FUSE_CAP_ATOMIC_O_TRUNC){
                   ~~~~  ^
s3fs.cpp:2677:36: error: use of undeclared identifier 'FUSE_CAP_ATOMIC_O_TRUNC'
  if((unsigned int)conn->capable & FUSE_CAP_ATOMIC_O_TRUNC){
                                   ^
s3fs.cpp:2678:12: error: no member named 'want' in 'fuse_conn_info'
     conn->want |= FUSE_CAP_ATOMIC_O_TRUNC;
     ~~~~  ^
s3fs.cpp:2678:20: error: use of undeclared identifier 'FUSE_CAP_ATOMIC_O_TRUNC'
     conn->want |= FUSE_CAP_ATOMIC_O_TRUNC;
                   ^
4 errors generated.
make[1]: *** [s3fs.o] Error 1
make[1]: *** Waiting for unfinished jobs....
```

**Comment by johntrandall** [comment]
Thanks for checking @DomT4 . Any advice on getting osxfuse and fuse4x to play nice side by side?

**Comment by DomT4** [comment]
If you're using <10.10 it's theoretically possible to stick ` fuse4x ` in a different directory, i.e ` ~/fuse4x ` and then use ` CFLAGS ` to point the formulae at that directory. If you're on 10.10 it's made more difficult by Yosemite's ban on unsigned kexts.

**Comment by jacknagel** [comment]
If I understand correctly, we are waiting on changes in osxfuse to make this possible. When they end up in a release, then we can make a change here. Until then, it is what it is. We aren't going to be able to make fuse4x and osxfuse install alongside each other.
